From 05fdab5434369be3fefef47d072403ca2c5b0aeb Mon Sep 17 00:00:00 2001
From: Matthijs van Duin <matthijsvanduin@gmail.com>
Date: Fri, 12 May 2017 00:55:26 +0200
Subject: [PATCH] sd-bus: fix c++ compatibility (#5941)

g++ annoyingly requires a non-empty struct-initializer to initialize all
struct members, in order of declaration.

Signed-off-by: Matthijs van Duin <matthijsvanduin@gmail.com>(cherry picked from commit cc9daff2289381d2fe7fb5ee7c3fa3435e8fc478)
---
 src/systemd/sd-bus-vtable.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/systemd/sd-bus-vtable.h b/src/systemd/sd-bus-vtable.h
index 3563a2b126..1e82cae038 100644
--- a/src/systemd/sd-bus-vtable.h
+++ b/src/systemd/sd-bus-vtable.h
@@ -131,6 +131,7 @@ struct sd_bus_vtable {
                         .member = _member,                              \
                         .signature = _signature,                        \
                         .get = _get,                                    \
+                        .set = NULL,                                    \
                         .offset = _offset,                              \
                     },                                                  \
                 },                                                      \
@@ -154,6 +155,9 @@ struct sd_bus_vtable {
 #define SD_BUS_VTABLE_END                                               \
         {                                                               \
                 .type = _SD_BUS_VTABLE_END,                             \
+                .flags = 0,                                             \
+                .x = {                                                  \
+                },                                                      \
         }
 
 _SD_END_DECLARATIONS;
